app/Models/User.php

```diff
@@ -220,10 +221,15 @@ public static function lookup($username_or_id, $lookup_type = null, $find_all =
         }
 
         if (!$find_all) {
-            $user = $user->where('user_type', 0)->where('user_warnings', 0);
+            $user = $user->where('user_type', 0);
         }
 
-        return $user->first();
+        $user = $user->first();
+        if (!$user || !$user->canView()) {
```

Comment:
it should be there. restricted/banned users shouldn't be returned anywhere system-wide.